src/ui/public/vis/editors/config/editor_config_providers.ts

```diff
@@ -0,0 +1,133 @@
+
+  public register(configProvider: EditorConfigProvider): void {
+    this.providers.add(configProvider);
+  }
+
+  public getConfigForAgg(
+    aggType: AggType,
+    indexPattern: IndexPattern,
+    aggConfig: AggConfig
+  ): EditorConfig {
+    const configs = Array.from(this.providers).map(provider =>
+      provider(aggType, indexPattern, aggConfig)
+    );
+    return this.mergeConfigs(configs);
+  }
+
+  private isBaseParam(config: EditorParamConfig): config is NumericIntervalParam {
+    return config.hasOwnProperty('base');
+  }
+
+  private isFixedParam(config: EditorParamConfig): config is FixedParam {
+    return config.hasOwnProperty('fixedValue');
+  }
+
+  private mergeHidden(current: EditorParamConfig, merged: EditorParamConfig): boolean {
+    return Boolean(current.hidden || merged.hidden);
+  }
+
+  private mergeFixedAndBase(
+    current: EditorParamConfig,
+    merged: EditorParamConfig,
+    paramName: string
+  ): { fixedValue?: any; base?: number } {
+    if (
+      this.isFixedParam(current) &&
+      this.isFixedParam(merged) &&
+      current.fixedValue !== merged.fixedValue
+    ) {
+      // In case multiple configurations provided a fixedValue, these must all be the same.
+      // If not we'll throw an error.
+      throw new Error(`Two EditorConfigProviders provided different fixed values for field ${paramName}:
+          ${merged.fixedValue} !== ${current.fixedValue}`);
+    } else if (
+      (this.isFixedParam(current) && this.isBaseParam(merged)) ||
+      (this.isBaseParam(current) && this.isFixedParam(merged))
+    ) {
+      // In case one config tries to set a fixed value and another setting a base value,
+      // we'll throw an error. This could be solved more elegantly, by allowing fixedValues
+      // that are the multiple of the specific base value, but since there is no use-case for that
+      // right now, this isn't implemented.
+      throw new Error(`Tried to provide a fixedValue and a base for param ${paramName}.`);
+    } else if (this.isBaseParam(current) && this.isBaseParam(merged)) {
+      // In case both had where interval values, just use the least common multiple between both interval
+      return {
+        base: leastCommonMultiple(current.base, merged.base),
+      };
+    } else {
+      // In this case we haven't had a fixed value of base for that param yet, we use the one specified
+      // in the current config
+      if (this.isFixedParam(current)) {
+        return {
+          fixedValue: current.fixedValue,
+        };
+      } else if (this.isBaseParam(current)) {
+        return {
+          base: current.base,
+        };
+      }
+
+      return {};
+    }
+  }
+
+  private mergeConfigs(configs: EditorConfig[]): EditorConfig {
+    return configs.reduce((output, conf) => {
+      Object.entries(conf).forEach(([paramName, paramConfig]) => {
+        if (!output[paramName]) {
+          output[paramName] = paramConfig;
```

Comment:
I know we're not mutating `paramConfig` anywhere in this code but is it possible someone might add code that adds properties to it in the future? If so, is it worth being defensive and just creating a shallow copy here? After all, we do that on line 122 so we may as well retain parity.

```js
output[paramName] = { ... paramConfig };
```